demosymfonyform/views/templates/admin/multipleForms.html.twig

```diff
@@ -0,0 +1,76 @@
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ *#}
+
+{% extends '@PrestaShop/Admin/layout.html.twig' %}
+{% form_theme demoConfigurationChoiceForm 'PrestaShopBundle:Admin/TwigTemplateForm:prestashop_ui_kit.html.twig' %}
+{% form_theme demoConfigurationOtherForm 'PrestaShopBundle:Admin/TwigTemplateForm:prestashop_ui_kit.html.twig' %}
+
+{% block content %}
+  {#  If you want form to use other url for saving you can set it here as path parameter(has to be route registered in routes.yml#}
+  {{ form_start(demoConfigurationChoiceForm, {attr : {class: 'form'}, action: path('demo_configuration_choices_form_save') }) }}
+  <div class="card">
+    <h3 class="card-header">
+      <i class="material-icons">settings</i> {{ 'Choice form types'|trans({}, 'Modules.DemoSymfonyForm.Admin') }}
+    </h3>
+    <div class="card-block row">
+      <div class="card-text">
+          {{ form_widget(demoConfigurationChoiceForm) }}
+      </div>
+    </div>
+    <div class="card-footer">
+      <div class="d-flex justify-content-end">
+        <button class="btn btn-primary float-right" id="demo-configuration-choices-form-save-button">
+          {{ 'Save'|trans({}, 'Admin.Actions') }}
+        </button>
+      </div>
+    </div>
+  </div>
+  {{ form_end(demoConfigurationChoiceForm) }}
+
+  {{ form_start(demoConfigurationOtherForm, {attr : {class: 'form'}, action: path('demo_configuration_other_form_save') }) }}
+  <div class="card">
+    <h3 class="card-header">
+      <i class="material-icons">settings</i> {{ 'Other form types'|trans({}, 'Modules.DemoSymfonyForm.Admin') }}
+    </h3>
+    <div class="card-block row">
+      <div class="card-text">
+        {{ form_widget(demoConfigurationOtherForm) }}
+      </div>
+    </div>
+    <div class="card-footer">
+      <div class="d-flex justify-content-end">
+        <button class="btn btn-primary float-right" id="save-button">
+          {{ 'Save'|trans({}, 'Admin.Actions') }}
+        </button>
+      </div>
+    </div>
+  </div>
+  {{ form_end(demoConfigurationOtherForm) }}
+{% endblock %}
+
+{% block javascripts %}
+  {{ parent() }}
+  <script src="{{ asset('../modules/demosymfonyform/views/js/multipleForms.js') }}"></script>
```

Comment:
Interesting 🤔  . Why the `../` ? what is the root path?

If we write this here, some people will copy it as the "how to include module asset" so I'm wondering if we can avoid the `../` . This should be avoided - if we can - in an example module.

Reply:
@matks I think the root path is `admin-dev` for example? so we have:
`admin-dev/../modules/[...]`

Reply:
overall I liked handling assets in legacy code better, it was easier to maintain with

```
$this->addJS(
    [
        'asset',
        'asset1',
        'asset2'
    ]
);
```

and 

`Media::addJsVar` etc.

Reply:
> overall I liked handling assets in legacy code better, it was easier to maintain with

If we were able to build an equivalent of `addJS()` but inside twig, would it be OK?

```
  {{ prestaShopAddModuleJS('demosymfonyform', 'views/js/js1.js') }}
  {{ prestaShopAddModuleJS('demosymfonyform', 'views/js/js2.js') }}
  {{ prestaShopAddModuleJS('demosymfonyform', 'views/js/main.js') }}
```

Reply:
@JevgenijVisockij Can I suggest we add a `@todo` here to add a helper function into the Core and then use it here (in another PR)?

Reply:
@JevgenijVisockij also see @zuk3975 example
https://github.com/PrestaShop/example-modules/pull/63/files#diff-4607570f92532573a9d82343e36d92bff974e3f92e115e2b789cbc66dfab891dR69

```
        $this->controller->addJS('modules/' . $this->name . '/views/js/orders-listing.js');
```
but maybe this can only be done inside a hook 😞

Reply:
addJS works for legacy controllers (or hook for that matter).

I don't remember seeing JS added in controller in migrated pages and this module should probably follow the same conventions that migrated pages do.

Reply:
We have 3 different cases when module wants to add JS:

1. Modules Symfony controller
2. Modules legacy controller
3. Extension of existing controllers

addJS works for cases 2 and 3 because we either have direct access to setMedia function, or we can access it via hook.

However it doesn't make sense for 1 because SymfonyControllers were never designed for that. In migrated pages we always add js in twigs block ``javascripts`` and in module I simply do the same thing.

I like ```{{ prestaShopAddModuleJS('demosymfonyform', 'views/js/js1.js') }} ``` approach the best. It still very close to how PrestaShop add javascript, just a little function to make pathing easier.

Reply:
So maybe we could have an issue "Add prestaShopAddModuleJS in twig". And I would add to todo to change the way js is added when we have this function?